src/gsp.py

```diff
@@ -38,20 +40,31 @@ def get_gsp_boundaries_from_eso_wgs84() -> gpd.GeoDataFrame:
     return boundaries
 
 
-@router.get("/forecast/one_gsp/{gsp_id}", response_model=Forecast)
+@router.get(
+    "/forecast/one_gsp/{gsp_id}",
+    response_model=Forecast,
+    dependencies=[Depends(get_auth_implicit_scheme())],
```

Comment:
By explicitly requiring auth on every endpoint it's very easy for us to create an endpoint where we forget about this requirement. If possible, I'd prefer to [implement the auth as a middleware](https://fastapi.tiangolo.com/tutorial/middleware/). We need to require auth on every single endpoint anyways, so if we implement it as a middleware that would be satisfied.

Reply:
Althought actually we might want to have one end point not authenticed, so we can easily do a quick check to see if they api is up and running